### ISSUE TITLE
dbt: when using "table" materialization, indexes aren't created

### ISSUE THREAD

**Issue by chuck-alt-delete** [MaterializeInc/materialize#20722]
### What version of Materialize are you using?

v0.61.0 (31314ef67)

## context

We have a user who uses the same dbt models in snowflake and materialize. For snowflake, they have to use `materialized='table'`. When running this in materialize, the materialized view is created but the indexes are not.

## repro

Here's a model:
```sql
{{ 
config(
    materialized = 'table',
    indexes=[
      {'columns': ['x']}
    ]
)}}

select 1 as x, 'a' as y union all
select 1 as x, 'b' as y union all
select 2 as x, 'c' as y 
```

The result is that the materialized view is created, but not the indexes.

Log file:
[mat_view_ix.log](https://github.com/MaterializeInc/materialize/files/12134853/mat_view_ix.log)

**Comment by morsapaes** [comment]
This is fixed in #20737. Once that is merged, we can release a new patch version (v1.5.1); this is ready in #20739.

**Comment by morsapaes** [comment]
For context, despite this being a bug, there is no good reason to use the `table` materialization explicitly, as a user. IIUC, in this case the user wants to keep parity between Snowflake and Materialuze dbt code, right? Since a materialized view is created, it could be less confusing and more maintainable to use `materializedview`, even if that breaks parity.

Supporting `table` as expected is blocked on supporting the `CREATE TABLE AS` syntax.
